blog: announcing localstack extensions (#65)
localstack/localstack.github.io: content/blog/2022-09-06-announcing-localstack-extensions/index.md

---
title: Announcing LocalStack Extensions
description: Announcing LocalStack Extensions
lead: Announcing LocalStack Extensions
date: 2022-09-06T5:51:39+05:30
lastmod: 2022-09-06T5:51:39+05:30
images: []
contributors: []
contributors: ["Harsh Mishra"]
tags: ['news']
---

With the LocalStack 1.0 General Availability going live last month, we announced LocalStack Extensions! LocalStack Extensions allow users to extend and customize LocalStack using pluggable Python distributions.

## Setting up LocalStack Extensions

LocalStack Extensions is part of our Pro offering. To get started with using LocalStack Extensions, first log in to your account using the LocalStack CLI:

```bash
$ localstack login
Please provide your login credentials below
Username: ...
```

After a log-in, you can get started using the LocalStack Extensions API:

```bash
$ localstack extensions --help

Usage: localstack extensions [OPTIONS] COMMAND [ARGS]...

  Manage LocalStack extensions (beta)

Options:
  --help  Show this message and exit.

Commands:

>   init       Initialize the LocalStack extensions environment

Reviewer: ```suggestion
Commands:
  dev        Developer tools for developing Localstack extensions
  init       Initialize the LocalStack extensions environment
```

  install    Install a LocalStack extension
  uninstall  Remove a LocalStack extension
```

## Using LocalStack Extensions?

> LocalStack Extensions is a Python application that runs together with LocalStack in the LocalStack container. LocalStack Extensions allows you to hook into different lifecycle phases of LocalStack and execute custom code or modify LocalStack’s HTTP gateway with custom routes and server-side code.

Reviewer: tiny nit:
```suggestion
LocalStack Extensions is a Python application that runs alongside the main process in the LocalStack Docker container. LocalStack Extensions allows you to hook into different lifecycle phases of LocalStack and execute custom code or modify LocalStack’s HTTP gateway with custom routes and server-side code.
```

We have the following extensions handy for our users to get started with:

- [Stripe LocalStack Extension](https://github.com/localstack/localstack-extensions/tree/main/stripe)
- [AWS Replicator Extension](https://github.com/localstack/localstack-extensions/tree/main/aws-replicator)

To install an extension, specify the name of the `pip` dependency that contains the extension. For example, for the official Stripe extension, you can either use the package distributed on PyPI:

```bash
localstack extensions install localstack-extensions-stripe
```

You can also install it directly from a Git repository:

```bash
localstack extensions install "git+https://github.com/localstack/localstack-extensions/#egg=localstack-extensions-stripe&subdirectory=stripe"
```

## Creating LocalStack Extensions

We invite developers using LocalStack to mock and emulate AWS infrastructure locally to help us build an ecosystem around LocalStack Extensions. LocalStack Extensions can be created using our core Extensions API in our core codebase.

{{< img src="localstack-extensions.png" >}}

> To create a new LocalStack Extension, you can use our Extensions CLI to access our developer commands that allows you to create new Extensions, and toggle local development mode for Extensions. With the developer mode toggled on, Extensions can be mounted into the LocalStack container hence you don't need to re-install them every time you change something.

Reviewer: ```suggestion
To create a new LocalStack Extension, you can use our Extensions CLI to access our developer commands that allows you to create new Extensions, and toggle local development mode for Extensions. With the developer mode toggled on, Extensions can be mounted into the LocalStack container, hence you don't need to re-install them every time you change something.
```
